deployer/deployer-services/src/main/java/io/hyscale/deployer/services/deployer/impl/KubernetesDeployer.java

```diff
@@ -484,5 +487,16 @@ private void deltaWait(long milliSeconds) {
         logger.debug("Replicas info:: unhandled case, pod owner: {}", podOwner);
         return K8sReplicaUtil.getReplicaInfo(podList);
     }
+
+    @Override
+    public List<AppMetadata> getAppsMetadata(AuthConfig authConfig) throws HyscaleException {
+        ApiClient apiClient = clientProvider.get((K8sAuthorisation) authConfig);
+
+        V1PodHandler podHandler = (V1PodHandler) ResourceHandlers.getHandlerOf(ResourceKind.POD.getKind());
+        
+        List<V1Pod> podList  = podHandler.getPodsForAllNamespaces(apiClient);
+        
+        return K8sDeployerUtil.getAppMetadata(podList);
```

Comment:
Can we make the appmeta building happening in a relevant class wrt to SRP like AppMetaDataBuilder